extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesAndWorkerTaskRunner.java

```diff
@@ -101,10 +103,19 @@ public void unregisterListener(String listenerId)
   @Override
   public ListenableFuture<TaskStatus> run(Task task)
   {
-    if (kubernetesAndWorkerTaskRunnerConfig.isSendAllTasksToWorkerTaskRunner()) {
-      return workerTaskRunner.run(task);
+    RunnerStrategy.RunnerType runnerType = runnerStrategy.getRunnerTypeForTask(task);
+    if (runnerType != null) {
+      if (RunnerStrategy.RunnerType.KUBERNETES_RUNNER_TYPE.equals(runnerType)) {
+        return kubernetesTaskRunner.run(task);
+      } else {
+        return workerTaskRunner.run(task);
+      }
     } else {
-      return kubernetesTaskRunner.run(task);
+      throw DruidException.defensive()
+                          .build(
+                              "Wrong runner type configured for task type [%s]",
```

Comment:
shouldn't this be more like "no runner type configured for task type?

Reply:
This is to handle the config like `index_kafka: wrong_runner_type`, in this case the value is wrong.
I think `no runner type configured` means different thing, as default runner is alway there, if `no runner type configured for task type` it should use the default one.